include/i2c.h

```diff
@@ -175,6 +204,105 @@ static inline int _impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+/**
+ * @brief Attaches the provided funcs and address as Slave device
+ *
+ * This routine starts the I2C Controller in Slave mode and uses
```

Comment:
I would write here: 

Enable I2C slave mode for the 'dev' I2C bus driver. The I2C slave will be registered at the address provided as 'address' parameter. Addressing mode - 7 or 10 bit - depends on the current configuration of the I2C bus driver. Any I2C bus events related to the slave mode will be passed onto I2C slave device driver via a set of callback functions provided as 'funcs' parameter.

Most of the existing hardware allows simultaneous support for master and slave mode. This is however not guaranteed.

Reply:
ok